sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsMeta.scala

```diff
@@ -650,8 +651,25 @@ abstract class SparkPlanMeta[INPUT <: SparkPlan](plan: INPUT,
         !childPlans.exists(_.supportsColumnar) &&
         (plan.conf.adaptiveExecutionEnabled ||
         !parent.exists(_.supportsColumnar))) {
+      // Some platforms can present a plan where the root of the plan is a shuffle followed by
+      // an AdaptiveSparkPlanExec. If it looks like the AdaptiveSparkPlanExec will end up
+      // columnar than this shuffle should be columnar as well.
+      val shuffle = wrapped.asInstanceOf[ShuffleExchangeExec]
+      val isChildColumnar = shuffle.child match {
+        case adaptiveExec: AdaptiveSparkPlanExec if parent.isEmpty =>
+          val aqePlan = adaptiveExec.executedPlan
+          if (aqePlan.supportsColumnar) {
```

Comment:
So what happens if all we have is a parquet read followed by an AdaptiveSparkPlanExec and then the shuffle? The `aqePlan` would support columnar, but it would not be on the GPU, at least not yet, and it is not guaranteed to be.

I guess this is a similar question around line 653. Is saying that the output data is columnar good enough, or do we want to know that the data is on the GPU?